tiledb/sm/array_schema/domain.cc

```diff
@@ -282,7 +275,7 @@ Status Domain::split_subarray_global(
     } else {
       s1[2 * i] = s[2 * i];
       s1[2 * i + 1] =
-          s1[2 * i] + MAX(1, floor(tiles_apart / 2)) * tile_extents[i];
+          s1[2 * i] + std::max(1.0, floor(tiles_apart / 2)) * tile_extents[i];
```

Comment:
One more thing, sorry. `s1` is of type `T`. If `T` was a non-floating point, this line would never perform any floating point conversions. This change now implies that `max` returns a floating point, but the existing macro did not. To preserve existing behavior, please use `std::max<T>()` here and leave the first argument as `1`.

Reply:
You're right. I changed that way. Thanks a lot!